pkg/list/values/extract.go

```diff
@@ -0,0 +1,308 @@
+package values
+
+import (
+	"fmt"
+	"strconv"
+	"strings"
+
+	"github.com/cloudposse/atmos/pkg/list/errors"
+	"github.com/cloudposse/atmos/pkg/utils"
+)
+
+// handleSpecialComponent processes special components like settings and metadata.
+func handleSpecialComponent(stack map[string]interface{}, component string) (map[string]interface{}, bool) {
+	// First check if the component exists at the top level
+	if section, ok := stack[component].(map[string]interface{}); ok {
+		return section, true
+	}
+
+	// If not found at the top level and component is "settings", look for it in components
+	if component == "settings" {
+		return extractSettingsFromComponents(stack)
+	}
+
+	return nil, false
+}
+
+// extractSettingsFromComponents extracts settings from terraform components.
+func extractSettingsFromComponents(stack map[string]interface{}) (map[string]interface{}, bool) {
+	allSettings := make(map[string]interface{})
+
+	// Try to navigate to terraform components
+	components, ok := stack["components"].(map[string]interface{})
+	if !ok {
+		return nil, false
+	}
+
+	terraform, ok := components["terraform"].(map[string]interface{})
+	if !ok {
+		return nil, false
+	}
+
+	// Collect settings from all terraform components
+	for componentName, componentData := range terraform {
+		if settings := extractComponentSettings(componentData); settings != nil {
+			allSettings[componentName] = settings
+		}
+	}
+
+	// Return all settings if we found any
+	if len(allSettings) > 0 {
+		return allSettings, true
+	}
+
+	return nil, false
+}
+
+// extractComponentSettings extracts settings from a component.
+func extractComponentSettings(componentData interface{}) interface{} {
```

Comment:
I don't understand the difference between `extractComponentSettings` and `extractSettingsFromComponents`